twoliter/embedded/rpm2img

```diff
@@ -192,6 +193,30 @@ if [[ "${INVENTORY_COUNT}" -ne "${RPM_COUNT}" ]]; then
   exit 1
 fi
 
+# Get the core kit version from external kit metadata.
+CORE_KIT_VERSION=$(jq --raw-output '.kit[]|select(.name == "bottlerocket-core-kit")|.version' $EXTERNAL_KIT_METADATA_PATH)
+CORE_KIT_VENDOR=$(jq --raw-output '.kit[]|select(.name == "bottlerocket-core-kit")|.vendor' $EXTERNAL_KIT_METADATA_PATH)
+
+# Query the core kit for package names. These names will be
+# compared to installed package names to determine what software
+# from the kit has been installed on this variant.
+CORE_KIT_INVENTORY_QUERY="\{\"Name\":\"%{NAME}\"\}"
+# shellcheck disable=SC2312 # Array is validated elsewhere.
+CORE_KIT_PATH="/host/build/external-kits/${CORE_KIT_VENDOR}/bottlerocket-core-kit/${ARCH}"
+mapfile -t all_core_kit_rpms <<<"$(dnf --repofrompath core-kit,$CORE_KIT_PATH --repo=core-kit repoquery --queryformat $CORE_KIT_INVENTORY_QUERY | sed 's/\\//g')"
+# Remove the 'bottlerocket-' prefix from package names.
+INVENTORY_DATA_KIT="$(jq --arg PKG_PREFIX "bottlerocket-" \
+  '(.Name) |= sub($PKG_PREFIX; "")' <<<"${INVENTORY_DATA_KIT}")"
```

Comment:
The testing data doesn't align with my understanding of what's happening here, so I'm probably wrong -- but it seems to me that `INVENTORY_DATA_KIT` isn't set at this point in the script?

Reply:
I didn't understand how this could work either. Overall the code is very hard to follow.

Reply:
Thanks for the callout, in  ad7af58 to ffd809e I had erroneously removed 

```
INVENTORY_DATA_KIT="$(jq --raw-output . <<<"${all_core_kit_rpms[@]}")"
```

The testing data was from a build before that change.

I'll add it back in. as far as the clarify of the code, I can't say I'm thrilled with its readability and I can work on that. I'll put out a rev that attempts to clarify things in comments.